tests/components/google/test_calendar.py

```diff
@@ -252,13 +255,70 @@ async def test_all_day_offset_event(hass, mock_events_list_items, component_setu
     }
 
 
-async def test_update_error(hass, calendar_resource, component_setup):
-    """Test that the calendar handles a server error."""
-    calendar_resource.return_value.get = Mock(
-        side_effect=httplib2.ServerNotFoundError("unit test")
-    )
-    assert await component_setup()
+async def test_update_error(
+    hass, calendar_resource, component_setup, test_api_calendar
+):
+    """Test that the calendar update handles a server error."""
+
+    now = dt_util.now()
+    with patch("homeassistant.components.google.api.google_discovery.build") as mock:
+        mock.return_value.calendarList.return_value.list.return_value.execute.return_value = {
+            "items": [test_api_calendar]
+        }
+        mock.return_value.events.return_value.list.return_value.execute.return_value = {
+            "items": [
+                {
+                    **TEST_EVENT,
+                    "start": {
+                        "dateTime": (now + datetime.timedelta(minutes=-30)).isoformat()
+                    },
+                    "end": {
+                        "dateTime": (now + datetime.timedelta(minutes=30)).isoformat()
+                    },
+                }
+            ]
+        }
+        assert await component_setup()
+
+    state = hass.states.get(TEST_ENTITY)
+    assert state.name == TEST_ENTITY_NAME
+    assert state.state == "on"
 
+    # Advance time to avoid throttling
+    now += datetime.timedelta(minutes=30)
+    with patch(
+        "homeassistant.components.google.api.google_discovery.build",
+        side_effect=httplib2.ServerNotFoundError("unit test"),
+    ), patch("homeassistant.util.utcnow", return_value=now):
+        await hass.helpers.entity_component.async_update_entity(TEST_ENTITY)
```

Comment:
If the entities are polling its better to fire a time change event to have the entities update.
https://github.com/home-assistant/core/blob/74c339eb7b70737b5c6d5a1325294002fe979a2b/tests/common.py#L380-L383


Side note: Using `hass.helpers` instead of importing names is legacy.

Reply:
Updated to use `async_fire_time_changed`

Reply:
I think the commit wasn't pushed, right?